optimum/onnxruntime/optimization.py

```diff
@@ -0,0 +1,274 @@
+            return "hidden_size"
+    
+    @staticmethod
+    def update_model(model_type:str, num_heads:str, hidden_size:str):
+        OnnxConfigManager.__conf[model_type] = {"num_heads":num_heads, "hidden_size":hidden_size}
+
+    @staticmethod
+    def remove_model(model_type:str):
+        try:
+            OnnxConfigManager.__conf.pop(model_type)
+        except KeyError:
+            print(f"{model_type} undefined in the configuration")
+    
+    @staticmethod
+    def check_supported_model(model_type:str) -> bool:
+        has_model = model_type in OnnxConfigManager.__conf.keys()
+        return has_model
+
+
+class ORTOptimizer:
+    def __init__(
+        self,
+        model_name_or_path: str,
+        ort_config: Union[str, ORTConfig],
+        feature: str = "default",
+        **kwargs
+    ):
+        """
+        Args:
+            model_name_or_path (:obj:`str`):
+                Repository name in the Hugging Face Hub or path to a local directory hosting the model.
+            ort_config (:obj:`Union[ORTConfig, str]`):
+                Configuration file containing all the information related to the model optimization and quantization.
+                Can be either:
+                    - an instance of the class :class:`ORTConfig`,
+                    - a string valid as input to :func:`ORTConfig.from_pretrained`.
+            feature (:obj:`str`):
+                Feature used when exporting the model.
+            data_files (:obj:`str`, `optional`):
+                Path to source data files.
+            cache_dir (:obj:`str`, `optional`):
+                Path to a directory in which a downloaded configuration should be cached if the standard cache should
+                not be used.
+            force_download (:obj:`bool`, `optional`, defaults to :obj:`False`):
+                Whether or not to force to (re-)download the configuration files and override the cached versions if
+                they exist.
+            resume_download (:obj:`bool`, `optional`, defaults to :obj:`False`):
+                Whether or not to delete incompletely received file. Attempts to resume the download if such a file
+                exists.
+            revision(:obj:`str`, `optional`):
+                The specific version to use. It can be a branch name, a tag name, or a commit id, since we use a
+                git-based system for storing models and other artifacts on huggingface.co, so ``revision`` can be any
+                identifier allowed by git.
+        """
+        config_kwargs_default = [
+            ("cache_dir", None),
+            ("force_download", False),
+            ("resume_download", False),
+            ("revision", None),
+        ]
+        config_kwargs = {name: kwargs.get(name, default_value) for (name, default_value) in config_kwargs_default}
+        self.cache_dir = config_kwargs.get("cache_dir")
```

Comment:
`self.cache_dir` not used so can be removed

Reply:
✅